corehq/apps/locations/forms.py

```diff
@@ -337,6 +304,152 @@ def save(self, instance=None, commit=True):
+
+        make_user = bound_data and (LocationType.objects
+                                   .get(domain=self.domain, name=bound_data['location_type'])
+                                   .has_user)
+        self.include_user_forms = is_new and (
+            (bound_data is None) or
+            make_user
+        )
+
+        if self.include_user_forms:
+            self.user_form = self._get_user_form(bound_data)
+            self.custom_user_data = self._get_custom_user_data(bound_data)
+            self.forms = [self.location_form, self.custom_location_data,
+                          self.user_form, self.custom_user_data]
+        else:
+            self.forms = [self.location_form, self.custom_location_data]
+
+    @memoized
+    def is_valid(self):
+        # Trigger the clean methods of each form
+        for form in self.forms:
+            form.errors
+        self._send_clean_signals()
+        return all(form.is_valid() for form in self.forms)
+
+    def _send_clean_signals(self):
+        clean_location.send(
+            self.__class__.__name__,
+            domain=self.domain,
+            request_user=self.request_user,
+            location=self.location,
+            forms={self.location_form.__class__.__name__: self.location_form},
+        )
+        if self.include_user_forms:
+            clean_commcare_user.send(
+                'MobileWorkerListView.create_mobile_worker',
+                domain=self.domain,
+                request_user=self.request_user,
+                user=self.user,
+                forms={
+                    self.user_form.__class__.__name__: self.user_form,
+                    self.custom_user_data.__class__.__name__: self.custom_user_data,
+                }
+            )
+
+    def save(self):
+        if not self.is_valid():
+            raise ValueError('Form is not valid')
+
+        if self.include_user_forms:
+            self.user.save()
+            self.location_form.location.user_id = self.user._id
+            location_data = self.custom_location_data.get_data_to_save()
+            location = self.location_form.save(metadata=location_data)
+            self.user.user_location_id = location.location_id
+            self.user.set_location(location)
+        else:
+            location_data = self.custom_location_data.get_data_to_save()
+            location = self.location_form.save(metadata=location_data)
+
+    @property
+    @memoized
+    def user(self):
+        user_data = self.custom_user_data.get_data_to_save()
+        username = self.user_form.cleaned_data['username']
+        password = self.user_form.cleaned_data['password']
+        first_name = self.user_form.cleaned_data['first_name']
+        last_name = self.user_form.cleaned_data['last_name']
+
+        return CommCareUser.create(
+            self.domain,
+            username,
+            password,
+            device_id="Generated from HQ",
```

Comment:
Actually I don't know what this is used for, or if it is used at all.  I just copied that form the NewMobileWorkerView.